Python/pywarpx/_libwarpx.py

```diff
@@ -882,22 +882,24 @@ def get_particle_z(self, species_name, level=0):
     def get_particle_id(self, species_name, level=0):
         '''
 
-        Return a list of numpy arrays containing the particle 'id'
-        positions on each tile.
+        Return a list of numpy arrays containing the particle id
+        numbers on each tile.
 
         '''
+        from .amrex_particle_id_utils import get_id
         structs = self.get_particle_structs(species_name, level)
-        return [struct['id'] for struct in structs]
+        return [get_id(struct['idcpu']) for struct in structs]
```

Comment:
Wouldn't it be  simpler if you skipped the extension, implemented a clean C++ version in WarpX as a helper function and just exposed that to Python?

Reply:
Side note: in pyAMReX, we declutter id and cpu by reinterpreting the idcpu as two 32bit numpy ints.